ons_alpha/private_media/middleware.py

```diff
@@ -0,0 +1,36 @@
+from django.shortcuts import redirect
+from wagtail.admin import messages
+
+
+class PreventRootCollectionChangesMiddleware:
+    """
+    A custom Middleware that prevents editing or deletion of the special 'Private' and
+    'Public' collections used by this project to restrict access to private media.
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        return self.get_response(request)
+
+    def process_view(self, request, view_func, view_args, view_kwargs):  # pylint: disable=unused-argument
```

Comment:
Question: Could this be done with another monkey-patch, rather than needing a complete middleware for a few small cases.

Reply:
Possibly, but I do really hate the monkey-patching too.... It's way way too easy for things to break in odd, difficult to pin-down ways. Try fudging one of the imports in a patch and see what happens :( Middleware is definitely the lesser of two evils, I think